modules/single-node-asg/main.tf

```diff
@@ -52,37 +52,69 @@ module "service-data" {
   iam_instance_profile_role_name = module.instance_profile.iam_role_name
 }
 
+resource "aws_eip" "eip" {
+  count = var.assign_eip ? 1 : 0
+}
+
+resource "aws_iam_role_policy_attachment" "associate_eip" {
+  role       = module.instance_profile.iam_role_name
+  policy_arn = aws_iam_policy.associate_eip_policy.arn
+}
+
+resource "aws_iam_policy" "associate_eip_policy" {
+  name   = "associate_address"
+  policy = data.aws_iam_policy_document.associate_eip_policy_doc.json
+}
+
+data "aws_iam_policy_document" "associate_eip_policy_doc" {
+  statement {
+    sid    = ""
+    effect = "Allow"
+    actions = [
+      "ec2:AssociateAddress"
+    ]
+    resources = ["*"]
+  }
+}
+
 # Create an ASG with just 1 EC2 instance
 module "server" {
   source = "../asg"
 
-  ami       = var.ami
-  elb_names = var.load_balancers
-  key_name  = var.key_name
+  ami                = var.ami
+  elb_names          = var.load_balancers
+  key_name           = var.key_name
   # The IAM Instance Profile w/ attach_ebs role
-  iam_profile   = module.instance_profile.iam_profile_id
-  instance_type = var.instance_type
-  # 1 EC2 instance <> 1 EBS volume 
-  max_nodes       = 1
-  min_nodes       = 1
-  placement_group = var.placement_group
-  public_ip       = var.public_ip
+  iam_profile        = module.instance_profile.iam_profile_id
+  instance_type      = var.instance_type
+  # 1 EC2 instance <> 1 EBS volume
+  max_nodes          = 1
+  min_nodes          = 1
+  placement_group    = var.placement_group
+  public_ip          = var.public_ip
   # the prefix and suffix names are combined in
   # the `asg` module to create the full name
-  name_prefix = var.name_prefix
-  name_suffix = "${var.name_suffix}-${local.az}"
-
+  name_prefix        = var.name_prefix
+  name_suffix        = "${var.name_suffix}-${local.az}"
   root_volume_type   = var.root_volume_type
   root_volume_size   = var.root_volume_size
   security_group_ids = var.security_group_ids
   subnet_ids         = [var.subnet_id]
 
   user_data = <<END_INIT
 #!/bin/bash
+<<<<<<< HEAD
 # exec > /tmp/init.log
 # exec 2> /tmp/init-err.log
 # set -x
+=======
+apt update
+>>>>>>> 2b82522... New function: single-node-asg module supports binding EIP by itself.
```

Comment:
@Magicloud what about resolving conflicts?

Reply:
Ah, I missed this one.